LCM/scripts/RemoveModule.py

```diff
@@ -4,124 +4,187 @@
-        elif os.path.isfile(moduleDir + "/" + curFile):
-            os.remove(moduleDir + "/" + curFile)
-        elif os.path.isdir(moduleDir + "/" + curFile):
-            RemoveMirroredDirectory(moduleDir + "/" + curFile, omiDir + "/" + curFile)
-            os.rmdir(moduleDir + "/" + curFile)
-            if len(os.listdir(omiDir + "/" + curFile)) == 0:
-                os.rmdir(omiDir + "/" + curFile)
-
-        if os.path.islink(omiDir + "/" + curFile):
-            os.path.unlink(omiDir + "/" + curFile)
-        elif os.path.isfile(omiDir + "/" + curFile):
-            os.remove(omiDir + "/" + curFile)
-
-if len(sys.argv) != 2:
-    usage()
-
-moduleName = sys.argv[1]
-omi_bindir = "<CONFIG_BINDIR>"
-omi_libdir = "<CONFIG_LIBDIR>"
-omi_sysconfdir = "<CONFIG_SYSCONFDIR>"
-baseModulePath = "<DSC_MODULES_PATH>"
-baseScriptPath = "<DSC_SCRIPT_PATH>"
-modulePath = baseModulePath + "/" + moduleName
-omiregister_dir = "/etc/opt/omi/conf/omiregister"
-
-if not os.path.isdir(modulePath):
-    print("Error: unable to find installed module " + moduleName + " at " + modulePath)
-    sys.exit(1)
+def printVerboseMessage(message):
+    verboseMessage = "VERBOSE from RemoveModule.py: " + message
+    print(verboseMessage)
 
-print("Removing module " + moduleName)
+def getPlatformArchitectureFolderName():
+    platformArchitecture = platform.architecture()
+    if len(platformArchitecture) != 2:
+        exitWithError("The python function platform.architecture() failed to return a valid tuple. Cannot detect if this system has an x64 or an x86 architecture.")
 
-# special section for nxOMSAutomationWorker module
-# the Linux Hybrid worker and manager needs to be killed before the module is removed
-# also a good idea to remove the state and working directory
-if moduleName == "nxOMSAutomationWorker":
-    state_dir_path = "/var/opt/microsoft/omsagent/state/automationworker"
-    working_dir_path = "/var/opt/microsoft/omsagent/run/automationworker"
-    automation_user = "nxautomation"
+    if (platformArchitecture[0] == "64bit"):
+        platformArchitectureFolderName = "x64"
+    else:
+        platformArchitectureFolderName = "x86"
 
-    # invoke deregister when it becomes available
-    # Kill all processes running as nxautomation
-    # In some cases, there might be no processes to kill, and the return code of the following command might be non-zero.
-    # We make a best attempt to terminate the processes and don't care about its return code
-    subprocess.call(["sudo", "pkill", "-u", automation_user, ".*"])
+    return platformArchitectureFolderName
 
-    # remove directories
-    shutil.rmtree(state_dir_path, ignore_errors=True)
-    shutil.rmtree(working_dir_path, ignore_errors=True)
+def regenerateDscPythonScriptInitFiles():
+    regenerateInitFilesScriptPath = join(helperlib.DSC_SCRIPT_PATH, 'RegenerateInitFiles.py')
+    regenerateInitFilesResult = subprocess.call(regenerateInitFilesScriptPath)
```

Comment:
> regenerateInitFilesResult = subprocess.call(regenerateInitFilesScriptPath) [](start = 3, length = 75)

This creates a new process to run this script. I am wondering why we need to create new process for this can we not call one python script from another and execute it in same process ? we make such calls all over the space and see errors around these calls because of running out of resources. may be better way is not to create new process to execute this.  what do you think ?

Reply:
This is a fairly common way to run a Python file from another Python file.
Another correct way we could do it is to import RegenerateInitFiles.py and then run a function from that file. But that files is not set up for that right now, so it would require changing that file as well.

But this is not related to my change for the side by side fix. My change merely moves this code that was already there in this file into a function for easier reference. I'm not sure what changing the processes will do in the 3 different versions of python we support and I don't want to change files that don't need to be changed for the fix. I do agree that this change should be made, but it should be in a separate PR that is not attached to the SxS fix.

Reply:
I agree we need rewrite of some of these python scripts to start running them in same process. Its just that we create too many processes and can randomly fail if limit is being reached.